### PR TITLE
Fix bug where ensure-migrations-table-exists fails if schema is included

### PR DESCRIPTION
**Pull request by deanmorin**
in table name

Fixes https://github.com/weavejester/ragtime/issues/117

May also be related to https://github.com/weavejester/ragtime/issues/80

### CODE REVIEW

File: jdbc/src/ragtime/jdbc.clj [hunk 1]
```diff
@@ -16,20 +16,29 @@
...
-      (get-table-names* conn))))
+      (get-table-metadata* conn))))
+
+(defn- has-schema?
+  [table-name]
+  (re-find #"\." table-name))
 
 (defn- table-exists? [db-spec ^String table-name]
-  (some #(.equalsIgnoreCase table-name %) (get-table-names db-spec)))
+  (some (fn [row]
```
Comment (weavejester) [inline jdbc/src/ragtime/jdbc.clj]: Let's change this part to:

```clojure
(defn- table-exists? [db-spec table-name]
  (some (partial metadata-matches-table? table-name)
        (get-table-metadata db-spec))
```

And then write:

```clojure
(defn metadata-matches-table? [^String table-name metadata]
   (.equalsIgnoreCase table-name
                      (if (.contains table-name ".")
                        (str (:table_schem metadata) "." (:table_name metadata))
                        (:table_name metadata)))
```

Which means we can get rid of `has-schema?`

**Comment by deanmorin** [comment]
That is a bit nicer. Made those changes (also made `metadata-matches-table?` private) and squashed the change. Happy to make any other changes

**Comment by weavejester** [comment]
Thanks! Can you also change the commit message to shorten the summary message? I follow [the seven rules](https://chris.beams.io/posts/git-commit/#seven-rules) for my messages.

So something more like:

    Fix table creation when table has schema

    Fixes #117.

**Comment by deanmorin** [comment]
Done

**Comment by iku000888** [comment]
@weavejester Would be really nice if this got released. Is there anything that is blocking a release?

**Comment by weavejester** [comment]
No, this just got lost in my todo list. I've released 0.7.2 now, @iku000888.

**Comment by iku000888** [comment]
Thanks!

**Pull request merged**